url913d8566d1_AnsBalin_rbn_tgz-pJ8-MainJ8/src/Main.java

```diff
@@ -0,0 +1,1121 @@
+    boolean containsMon, containsDi;
+    
+    containsMon=false;
+    containsDi=false;
+    int ABIndex, CIndex, ACIndex;
+    double A1, A2, A3;
+    ABIndex=0;
+    CIndex=0;
+    ACIndex=0;
+    
+    // Check reaction for form: AC + B --> AB + C
+    if( (products.size() == 2) && (products.get(0).getID() != reactants.get(0).getID()) ){
+      
+      for(int i=0; i<products.size(); i++){
+        if(products.get(i).getSize()==1){containsMon=true; CIndex=i;}
+        if(products.get(i).getSize()==2){containsDi=true; ABIndex = i;}
+        if(reactants.get(i).getSize()==2){ACIndex = i;}
+      }
+      if(containsMon && containsDi){
+        
+        Molecule AB = new Molecule(products.get(ABIndex));
+        Molecule A = new Molecule(AB.getChildren(0));
+        Molecule B = new Molecule(AB.getChildren(1));
+        Molecule C = new Molecule(products.get(CIndex));
+        Molecule AC = new Molecule(reactants.get(ACIndex));
+        
+        // Get activation energy level A1 for reaction A + B -> AB
+        double cAB = (double) retrieve(AB).getCycleLength();
+        double aAB = retrieve(AB).getActivity();
+        double cA = (double) retrieve(A).getCycleLength();
+        double aA = retrieve(A).getActivity();
+        double cB = (double) retrieve(B).getCycleLength();
+        double aB = retrieve(B).getActivity();
+        
+        A1 = (cAB + cA + cB)/2 + (aAB + aA + aB)/3 - (cA + cB);
+        
+        
+        // Get compound activation energy level A2+A3 for reactions A + C -> AC and AC + B -> AB + C
+        double cC = (double) retrieve(C).getCycleLength();
+        double aC = retrieve(C).getActivity();
+        double cAC = (double) retrieve(AC).getCycleLength();
+        double aAC = retrieve(AC).getActivity();
+        
+        A2 = (cAC + cA + cC)/2 + (aAC + aA + aC)/3 - (cA + cC);
+        A3 = (cAC + cB + cAB + cC)/2 + (aAC + aB + aAB + aC)/4 - (cAC + cB);
+        
+        if((A2+A3 < A1) && (aC !=0)){
+          
+          
+          if( !(foundCatalysisProduct) ){
+            catalysisProductIndex = AB.getID();
+            foundCatalysisProduct = true;
+            
+          }
+          
+          /*System.out.printf("\n\tMolecule %d catalyses the reaction %d + %d --> %d\n", C.getID(), A.getID(), B.getID(), AB.getID());
+          System.out.printf("\t\tCyLen.\tActiv.\n");
+          System.out.printf("\tA\t%.2f\t%.2f\n", cA, aA);
+          System.out.printf("\tB\t%.2f\t%.2f\n", cB, aB);
+          System.out.printf("\tC\t%.2f\t%.2f\n", cC, aC);
+          System.out.printf("\tAB\t%.2f\t%.2f\n", cAB, aAB);
+          System.out.printf("\tAC\t%.2f\t%.2f\n", cAC, aAC);
+          System.out.printf("\tA1: %f\n", A1);
+          System.out.printf("\tA2+A3: %f\n", A2+A3);*/
+          
+        
+        }
+        
+        
+      }
+      
+    
+    
+    
+    }
+  
+  
+  }
+  
+  public void collide(Molecule A, Molecule B, boolean bspar){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 19. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.